pkg/util/application/conditions.go

```diff
@@ -1,13 +1,70 @@
 package application
 
 import (
+	"fmt"
+	"sort"
+	"strings"
+
 	coreV1 "k8s.io/api/core/v1"
 	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
-	"sort"
+	diffutil "github.com/bookingcom/shipper/pkg/util/diff"
 )
 
+type ConditionDiff struct {
+	cond1, cond2 *shipper.ApplicationCondition
+}
```

Comment:
I wonder if it isn't time we introduce an interface for conditions? Or, less preferred because of type safety, consolidate everything in a `Condition` type? Besides type safety, I see very little value in having to duplicate this type of code for every object we have.

Reply:
I assume, the early development of these conditions was inspired by the native k8s approach (e.g.: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2253). I wouldn't really challenge this approach of k8s and preserve it the way it is in shipper as well. From this perspective an interface looks like a decent alternative.